### PR TITLE
fix(table permission): index name displays as table name in sql windows check

### LINKED ISSUE #3341: [Bug]: When creating an index for a table without change permission, consider the index name as the table name
### ODC version

odc432

### OB version

*

### What happened?

image

### What did you expect to happen?

Just need the change permission of the original table

### How can we reproduce it (as minimally and precisely as possible)?

Create a table without change permission and create an index

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
image
The sql check window does not display the index
#### Which issue(s) this PR fixes:

Fixes #3341

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java [hunk 1]
```diff
@@ -196,6 +197,16 @@ private static class OBMySQLRelationFactorVisitor extends OBParserBaseVisitor<Re
 
         private final Set<DBSchemaIdentity> identities = new HashSet<>();
 
+        @Override
+        public RelationFactor visitCreate_index_stmt(Create_index_stmtContext ctx) {
+            for (int i = 0; i < ctx.getChildCount(); i++) {
+                if (i != 2) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java:203]: magic number, which is not acceptable~
btw, how about alter table add index statement

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java [hunk 2]
```diff
@@ -352,6 +365,17 @@ public RelationFactor visitRename_table_stmt(OBParser.Rename_table_stmtContext c
...
         }
 
+        @Override
+        public RelationFactor visitCreate_index_stmt(OBParser.Create_index_stmtContext ctx) {
+            addRelationFactor(OracleFromReferenceFactory.getRelationFactor(ctx.relation_factor()));
+            return null;
+        }
+
+        @Override
+        public RelationFactor visitAlter_index_stmt(OBParser.Alter_index_stmtContext ctx) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java:375]: we should consider  of alter index option

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java [hunk 3]
```diff
@@ -213,6 +215,17 @@ public RelationFactor visitRename_table_stmt(Rename_table_stmtContext ctx) {
...
         }
 
+        @Override
+        public RelationFactor visitCreate_index_stmt(Create_index_stmtContext ctx) {
+            addRelationFactor(MySQLFromReferenceFactory.getRelationFactor(ctx.relation_factor()));
+            return null;
+        }
+
+        @Override
+        public RelationFactor visitAlter_index_option(Alter_index_optionContext ctx) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java:225]: is it necessary?

Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java:225]: unnecessary, will remove it

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**